### ISSUE TITLE
Gradle 8.6 reports deprecation warnings with plugin version 5.2.0

### ISSUE THREAD

**Issue by bmeier-pros** [jfrog/artifactory-gradle-plugin#107]
### Describe the bug

After upgrading to plugin version 5.2.0, we started receiving deprecation warnings from Gradle 8.6 that were not present with version 5.1.14.  After downgrading back to 5.1.14, the deprecation warnings went away.

```
> Configure project :
The Project.getConvention() method has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.api.plugins.Convention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#deprecated_access_to_conventions
The org.gradle.util.ConfigureUtil type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#org_gradle_util_reports_deprecations
	at build_1tda8vsatwmul6fo6fllw61ss.run(build.gradle:159)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
Build file 'build.gradle': line 160
The org.gradle.util.ConfigureUtil type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#org_gradle_util_reports_deprecations
	at build_1tda8vsatwmul6fo6fllw61ss$_run_closure11.doCall$original(build.gradle:160)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
	at build_1tda8vsatwmul6fo6fllw61ss.run(build.gradle:159)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
Build file 'build.gradle': line 162
The org.gradle.util.ConfigureUtil type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#org_gradle_util_reports_deprecations
	at build_1tda8vsatwmul6fo6fllw61ss$_run_closure11$_closure31.doCall$original(build.gradle:162)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
	at build_1tda8vsatwmul6fo6fllw61ss$_run_closure11.doCall$original(build.gradle:160)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
Build file 'build.gradle': line 168
The org.gradle.util.ConfigureUtil type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#org_gradle_util_reports_deprecations
	at build_1tda8vsatwmul6fo6fllw61ss$_run_closure11$_closure31.doCall$original(build.gradle:168)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
	at build_1tda8vsatwmul6fo6fllw61ss$_run_closure11.doCall$original(build.gradle:160)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
Build file 'build.gradle': line 159
The org.gradle.api.plugins.Convention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/8.6/userguide/upgrading_version_8.html#deprecated_access_to_conventions
	at build_1tda8vsatwmul6fo6fllw61ss.run(build.gradle:159)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
```

### Current behavior

Deprecation warnings are present, a regression from the previous version.

### Reproduction steps

Artifactory configuration block in `build.gradle` file:
```
artifactory {
    publish {
        contextUrl = artifactoryUrl
        repository {
            repoKey = isRelease ? releaseKey : snapshotKey
            username = artifactory_user
            password = artifactory_password
        }
        defaults {
            publications ('ALL_PUBLICATIONS')
        }
    }
}
```

### Expected behavior

no depreciation warnings surfaced by Gradle

### Artifactory Gradle plugin version

5.2.0

### Operating system type and version

Linux 5.15.133.1-microsoft-standard-WSL2 #1 SMP Thu Oct 5 21:02:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### JFrog Artifactory version

N/A

### Gradle version

8.6

**Comment by yahavi** [comment]
@bmeier-pros 
Thank you for using the Gradle Artifactory plugin.

We stopped using `ConventionUtils` in version [5.1.10](https://github.com/jfrog/artifactory-gradle-plugin/releases/tag/5.1.10). Could you please try using the latest plugin version (currently 5.2.3) with the `--stacktrace` flag and share the output?

**Comment by bmeier-pros** [comment]
@yahavi - the plugin was fixed starting with version `5.2.1` .  I assume the 5.2.0 version was released without the port of the `5.1.10` change to remove that.  I'm currently using `5.2.3`, and it is working fine for our use case.
